overlord/configstate/configcore/utils.go

```diff
@@ -95,3 +95,28 @@ func updateKeyValueStream(r io.Reader, supportedConfigKeys map[string]bool, newC
 
 	return nil, nil
 }
+
+type filterFunc func(values, filtered map[string]interface{})
+
+func applyFilters(flagFilter func(f flags) filterFunc, values map[string]interface{}) (map[string]interface{}, []configHandler) {
```

Comment:
Should this be named `applyPreinstallFilters'?

Reply:
well there are two cases, one for the preinstall case, and one for the early case, as seen in 

https://github.com/snapcore/snapd/pull/10089/files#diff-8a536e8f5201c9b0d34c49ec4ff4e41f0d416b8d5fe84aa194f8cc0c03e5acb5R142-R144

so given with the fact there are two uses right now and also that configcore is a bit unwieldly in the number of places/times that it can be used from, I think it's best to be defensive here in case we do end up needing another time/place that needs filtering.